server/src/main/java/org/apache/druid/server/log/AbstractFileRequestLogger.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.server.log;
+
+import com.google.common.base.Throwables;
+import org.apache.druid.java.util.common.DateTimes;
+import org.apache.druid.java.util.common.concurrent.ScheduledExecutors;
+import org.apache.druid.java.util.common.guava.CloseQuietly;
+import org.joda.time.DateTime;
+import org.joda.time.Duration;
+import org.joda.time.MutableDateTime;
+import org.joda.time.chrono.ISOChronology;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ScheduledExecutorService;
+
+public abstract class AbstractFileRequestLogger
```

Comment:
I don't think it's necessary to split the RequestLogger implementations into native/SQL versions, it's enough that there's a new SqlRequestLogLine implementation.

If you want to have two different request loggers for SQL and native queries, I think would be better to bind the SQL provider to a different configuration parameter and inject the desired provider using annotations (maybe like how you can get a `@Coordinator` DruidLeaderClient and a separate `@IndexingService` instance)

Reply:
@gianm Do you have a strong preference on single request logger vs. separate loggers for SQL and native queries? I saw there was some discussion around that in the proposal, I'm personally fine with either approach

Reply:
Hmm, after thinking about it more, I think it's better to have everything go through one RequestLogger interface. Will write up the reasons why in a separate comment.